internal/data/trace.go

```diff
@@ -121,14 +121,24 @@ func NewSpan() *Span {
 	return &Span{orig: &otlptrace.Span{}}
 }
 
-// NewSpanSlice creates a slice of Spans that are correctly initialized.
-func NewSpanSlice(len int) []Span {
+// NewSpanSlice creates a slice of pointers to Spans that are correctly initialized.
+func NewSpanSlice(len int) []*Span {
+	// Slice for underlying data.
 	origs := make([]otlptrace.Span, len)
+
+	// Slice for wrappers.
 	wrappers := make([]Span, len)
+
+	// Slice for pointers to wrappers.
+	ptrs := make([]*Span, len, len)
```

Comment:
One of the two options:
1. Just ptrs := make([]*Span, len)
2. Use ptrs := make([]*Span, 0, len) then append on line 139. The advantage is that you don't need to initialize the pointers, but I think append does a check for the capacity all the calls, so probably better with the first version.

Reply:
It is a typo. Fixed.